staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -1442,9 +1442,11 @@ func TestConnectionResetByPeerIsRetried(t *testing.T) {
 }
 
 func TestCheckRetryHandles429And5xx(t *testing.T) {
+	var rwlock sync.RWMutex
 	count := 0
 	ch := make(chan struct{})
 	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		rwlock.Lock()
```

Comment:
```suggestion
		rwlock.Lock()
		defer rwlock.Unlock()
```

There is an early return in this function before the `Unlock` line might be hit, so it is possible that the code as written could return early while the lock is held. You can prevent that bug by calling `Unlock` in a `defer` statement immediately after locking.